topic_modeling/train_models/train_models.py

```diff
@@ -0,0 +1,166 @@
+import os
+import sys
+import glob
+import re
+import nltk
+import string
+import pickle
+from sklearn.feature_extraction.text import CountVectorizer
+from sklearn.decomposition import LatentDirichletAllocation
+from sklearn.feature_extraction import text
+from nltk.tokenize import word_tokenize
+from nltk.stem.porter import PorterStemmer
+from nltk.corpus import stopwords
+from nltk.stem.wordnet import WordNetLemmatizer
+
+nltk.download("stopwords")
+nltk.download("wordnet")
+nltk.download("punkt")
+
+stop = set(stopwords.words('english'))
+exclude = set(string.punctuation)
+lemma = WordNetLemmatizer()
+
+
+def print_top_words(model, feature_names, num_topics,
+                    output_dir, stemming_info, n_top_words=20):
+    myfile = open(os.path.join(
+            output_dir, '{0}_topics_{1}'.format(
+                num_topics, stemming_info)), 'w')
+    for topic_idx, topic in enumerate(model.components_):
+        message = "Topic #{}: ".format(topic_idx)
+        message += " ".join(
+            [feature_names[i]
+                for i in topic.argsort()[:-n_top_words - 1:-1]])
+        myfile.write("{}\n".format(message))
+    myfile.close()
+
+
+def preprocess_documents(doc, stemming=True):
+    porter = PorterStemmer()
+    s = doc.lower()
+    s = re.sub(r'([^\s\w]|_)+', ' ', s)
+    s = re.sub(r'\s+', ' ', s)  # gets rid of tabs
+    s = re.sub(r'[^A-Za-z0-9]+', ' ', s)
+    tokens = word_tokenize(s)
+    if stemming:
+        stemmed = [porter.stem(word) for word in tokens]
+        s = " ".join(stemmed)
+    return s
+
+
+def test_preprocess_documents():
+    assert preprocess_documents(
+            'Clean THIS up.,!@ stemming') == 'clean thi up stem'
+    assert preprocess_documents(
+            'Clean THIS up.,!@ stemming',
+            stemming=False) == 'clean this up stemming'
+
+
+def flag_stemming(stemming=True):
+    if stemming:
+        stemming_info = 'with_stemming'
+    else:
+        stemming_info = 'without_stemming'
+    return stemming_info
+
+
+# will want an optional parameter for the documents.
+# can set it here
+def train_models(topics, stemming=True):
+
+    # need two different ways to set directory to pass
+    # Travis CI (because .travis file is called from
+    # different directory)
+    current_file = os.getcwd().split('/')[-1]
+
+    if current_file == 'train_models':
+        base_dir = os.getcwd()
+        data_dir = os.path.join(
+            os.path.normpath(os.getcwd() + os.sep + os.pardir),
+            'scrape_training_data')
+    else:
+        base_dir = os.path.join(
+                os.getcwd(), 'topic_modeling', 'train_models')
```

Comment:
related to cmdline options -- eventually you might want to stop relying on being in a current dir and provide an option to define what is the top directory for the data(set)

Reply:
I totally agree. Planning on implementing this eventually (and generally making this more of a 'standalone' package, but want to prioritize being able to generate model comparisons for the purposes of this final project. Will rely on my specific file structure / training data for now. But do want to allow for users to point to their own training data eventually.